Sprint-1/implement/sum.test.js

```diff
@@ -9,28 +9,42 @@ E.g. sum(['hey', 10, 'hi', 60, 10]), target output: 80 (ignore any non-numerical
 const sum = require("./sum.js");
 
 // Acceptance Criteria:
-
-// Given an empty array
-// When passed to the sum function
-// Then it should return 0
-test.todo("given an empty array, returns 0")
-
-// Given an array with just one number
-// When passed to the sum function
-// Then it should return that number
-
-// Given an array containing negative numbers
-// When passed to the sum function
-// Then it should still return the correct total sum
-
-// Given an array with decimal/float numbers
-// When passed to the sum function
-// Then it should return the correct total sum
-
-// Given an array containing non-number values
-// When passed to the sum function
-// Then it should ignore the non-numerical values and return the sum of the numerical elements
-
-// Given an array with only non-number values
-// When passed to the sum function
-// Then it should return the least surprising value given how it behaves for all other inputs
+describe("", () => {
+  // Given an empty array
+  // When passed to the sum function
+  // Then it should return 0
+  //   test.todo("given an empty array, returns 0");
+  test("given an empty array, returns 0", () => {
+    expect(sum([])).toBe(0);
+  });
+  // Given an array with just one number
+  // When passed to the sum function
+  // Then it should return that number
+  test("given an array with just one number, returns that number", () => {
+    expect(sum([5])).toBe(5);
+  });
+  // Given an array containing negative numbers
+  // When passed to the sum function
+  // Then it should still return the correct total sum
+  test("given an array containing negative numbers, returns the correct total sum", () => {
+    expect(sum([-10, -20, 30])).toBe(0);
+  });
+  // Given an array with decimal/float numbers
+  // When passed to the sum function
+  // Then it should return the correct total sum
+  test("given an array with decimal/float numbers, returns the correct total sum", () => {
+    expect(sum([1.5, 2.5, 3])).toBe(7);
```

Comment:
Decimal numbers in most programming languages (including JS) are internally represented in "floating point number" format. Floating point arithmetic is not exact. For example, the result of `46.5678 - 46 === 0.5678` is false because `46.5678 - 46` only yield a value that is very close to `0.5678`.

So `expect(sum([1.1, 1.1, 1.1])).toEqual(3.3);` would probably fail.

Can you find a more appropriate way to test a value (that involves decimal number calculations) for "equality?

Reply:
test("given an array with decimal numbers, returns the correct total sum", () => {
  expect(sum([1.1, 1.1, 1.1])).toBeCloseTo(3.3, 5);
});